testdata/p4_16_samples/psa-random-bmv2.p4

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2020 Cornell University
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include "psa.p4"
+
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+struct empty_metadata_t {
+}
+
+struct metadata_t {
+}
+
+struct headers_t {
+    ethernet_t       ethernet;
+}
+
+parser IngressParserImpl(packet_in pkt,
+                         out headers_t hdr,
+                         inout metadata_t user_meta,
+                         in psa_ingress_parser_input_metadata_t istd,
+                         in empty_metadata_t resubmit_meta,
+                         in empty_metadata_t recirculate_meta)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        transition accept;
+    }
+}
+
+control cIngress(inout headers_t hdr,
+                 inout metadata_t user_meta,
+                 in    psa_ingress_input_metadata_t  istd,
+                 inout psa_ingress_output_metadata_t ostd)
+{
+    Random<bit<48>>(1, 4) rand;
+    apply {
+        hdr.ethernet.dstAddr = 2;
+        if (rand.read() < 5) {
+             hdr.ethernet.dstAddr = 3;
+        }
+        // To see random behavior, un-comment the line below this comment to see
+        // package coming out from different ports. 
+        // The line is currently commented so that STF test can pass.
+        // hdr.ethernet.dstAddr = rand.read();
```

Comment:
I checked the functionality of rand.read() manually: it is returning a random int in the range [1, 4]. 

However, I haven't been able to think of a way to check this functionality automatically with STF. This test program is testing `rand.read() < 5`. Comments/suggestions appreciated!

Reply:
@yunheL STF tests are not really a good method for testing this kind of thing, at least not via exact match comparisons between the expected and actual output packets.

You could tighten up the condition `(rand.read() < 5)` in your test program to something like this for a little bit of extra checking that things are going correctly, but I can't think of anything more than that off hand:

```
bit<48> r;

r = rand.read();
if ((r >= 1) && (r <= 4)) {
    hdr.ethernet.dstAddr = 3;
}
```